tests/unit/references/test_references_processors.py

```diff
@@ -0,0 +1,643 @@
+
+    expected = [
+        {
+            'reference': {
+                'label': 'Abe et al, 2008',
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_set_record():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+    ref = get_record_ref(1226234, 'literature')
+
+    builder.set_record(ref)
+
+    expected = [
+        {
+            'curated_relation': False,
+            'record': {
+                '$ref': 'http://localhost:5000/api/literature/1226234',
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_curate():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.curate()
+
+    expected = [
+        {'curated_relation': True},
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_set_texkey():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.set_texkey('Aaij:2016qlz')
+
+    expected = [
+        {
+            'reference': {
+                'texkey': 'Aaij:2016qlz',
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_add_title():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.add_title('The CMS experiment at the CERN LHC')
+
+    expected = [
+        {
+            'reference': {
+                'title': {
+                    'title': 'The CMS experiment at the CERN LHC',
+                }
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_add_misc():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.add_misc('[Erratum:')
+
+    expected = [
+        {
+            'reference': {
+                'misc': [
+                    '[Erratum:',
+                ],
+            },
+        },
+    ]
+    result = [builder.obj]
+
+    assert validate(result, subschema) is None
+    assert expected == result
+
+
+def test_add_raw_reference():
+    schema = load_schema('hep')
+    subschema = schema['properties']['references']
+
+    builder = ReferenceBuilder()
+
+    builder.add_raw_reference('Phys. Rev. C 80 (doi:10.1103/PhysRevC.80.044313)')
+
+    expected = [
+        {
+            'raw_refs': [
+                {
+                    'schema': 'text',
+                    'source': 'reference_builder',
```

Comment:
again this is not a source. I think `add_raw_reference` should take a `source` optional argument (or even `__init__`), like for the [`LiteratureBuilder`](https://github.com/inspirehep/inspire-schemas/blob/master/inspire_schemas/builders.py#L89)